packages/core/src/types.ts

```diff
@@ -366,16 +370,15 @@ export type TransitionConfigTarget<TContext, TEvent extends EventObject> =
 
 export type TransitionConfigOrTarget<
   TContext,
-  TEvent extends EventObject
+  TEvent extends EventObject,
+  TEventType extends TEvent['type'] = TEvent['type']
 > = SingleOrArray<
-  TransitionConfigTarget<TContext, TEvent> | TransitionConfig<TContext, TEvent>
+  | TransitionConfigTarget<TContext, TEvent>
+  | TransitionConfig<TContext, TEvent, TEventType>
 >;
 
-type TransitionsConfigMap<TContext, TEvent extends EventObject> = {
-  [K in TEvent['type']]?: TransitionConfigOrTarget<
-    TContext,
-    TEvent extends { type: K } ? TEvent : never
-  >;
+export type TransitionsConfigMap<TContext, TEvent extends EventObject> = {
+  [K in TEvent['type']]?: TransitionConfigOrTarget<TContext, TEvent, K>;
```

Comment:
Any particular reason behind this change? 🤔 seems a little bit stranger to pass `TEvent` and `K` around and compute at the end rather than pass a computed `TEvent` from here

is it about `TransitionConfigTarget` including a `StateNode` target? wouldn't it be better to allow there `any` at the `TEvent` position as we should be able to target any state node in the given machine config? As this is going to be removed in v5 anyway (targeting with getters etc) I'm not sure if it's worth complicating the codebase for its sake

Reply:
I think this is an accidental commit, thanks for catching.

Reply:
Oh, I just discovered one of the reasons for this change: the `raise()` event needs to be able to infer all possible events.